homeassistant/components/aladdin_connect/config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Config flow for Aladdin Connect cover integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aladdin_connect import AladdinConnectClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    acc = AladdinConnectClient(data["username"], data["password"])
+    try:
+        login = await hass.async_add_executor_job(acc.login)
+    except (TypeError, KeyError, NameError, ValueError) as ex:
+        raise CannotConnect from ex
+    else:
+        if not login:
+            raise InvalidAuth
+
+    return {"title": "Aladdin Connect"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Aladdin Connect."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+
+        else:
+            await self.async_set_unique_id(
+                user_input["username"].lower(), raise_on_progress=False
+            )
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(title=info["title"], data=user_input)
```

Comment:
If `validate_input` does not return anything you can specify the string here
```suggestion
            return self.async_create_entry(title="Aladdin Connect", data=user_input)
```